crates/matrix-sdk-crypto/src/gossiping/machine.rs

```diff
@@ -826,74 +829,71 @@ impl GossipMachine {
+            self.inner.store.get_device_from_curve_key(&secret.event.sender, sender_key).await?
         {
             // Only accept secrets from one of our own trusted devices.
             if device.user_id() == self.user_id() && device.is_verified() {
-                self.accept_secret(event, request, secret_name).await?;
+                self.accept_secret(secret, changes).await?;
             } else {
-                warn!(
-                    request_id = event.content.request_id.as_str(),
-                    secret_name = secret_name.as_ref(),
-                    "Received a m.secret.send event from another user or from \
-                    unverified device"
-                );
+                warn!("Received a m.secret.send event from another user or from unverified device");
             }
         } else {
-            warn!(
-                request_id = event.content.request_id.as_str(),
-                secret_name = secret_name.as_ref(),
-                "Received a m.secret.send event from an unknown device"
-            );
-            self.inner.store.mark_user_as_changed(&event.sender).await?;
+            warn!("Received a m.secret.send event from an unknown device");
+
+            self.inner.store.mark_user_as_changed(&secret.event.sender).await?;
         }
 
         Ok(())
     }
 
+    #[instrument(skip_all, fields(sender_key, sender = %event.sender, request_id = %event.content.request_id, secret_name))]
     pub async fn receive_secret_event(
         &self,
         sender_key: Curve25519PublicKey,
         event: &DecryptedSecretSendEvent,
+        changes: &mut Changes,
     ) -> Result<Option<SecretName>, CryptoStoreError> {
-        debug!(request_id = event.content.request_id.as_str(), "Received a m.secret.send event");
+        debug!("Received a m.secret.send event");
 
-        let maybe_request = self
-            .inner
-            .store
-            .get_outgoing_secret_requests(event.content.request_id.as_str().into())
-            .await?;
+        let request_id = &event.content.request_id;
 
-        Ok(if let Some(request) = maybe_request {
+        let name = if let Some(request) =
+            self.inner.store.get_outgoing_secret_requests(request_id).await?
+        {
             match &request.info {
                 SecretInfo::KeyRequest(_) => {
-                    warn!(
-                        request_id = event.content.request_id.as_str(),
-                        "Received a m.secret.send event but the request was for a room key"
-                    );
+                    warn!("Received a m.secret.send event but the request was for a room key");
 
                     None
                 }
                 SecretInfo::SecretRequest(secret_name) => {
-                    self.receive_secret(sender_key, event, &request, secret_name).await?;
+                    Span::current().record("secret_name", display(secret_name));
```

Comment:
```suggestion
                    Span::current().record("secret_name", debug(secret_name));
```

Probably the same thing here.